### PR TITLE
Feat(eos_cli_config_gen): Add TTL max hops in router bgp

### LINKED ISSUE #3424: Feat(eos_cli_config_gen): Add support for `neighbor <neighbor_ID> ttl maximum-hops <hop_number> ` for neighbors and peer-groups
### Enhancement summary

cf title

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Security measure for BGP

### Describe the solution you would like

```
router_bgp:
  neighbors:
    - ip_address: 42.42.42.42
       ttl_maximum_hops: <int>;  0-254 
  peer_groups:
    - name: MyAwesomePG-42
       ttl_maximum_hops: <int>;  0-254 
```

host1(config-router-bgp)#neighbor 2.2.2.2 ttl maximum-hops ?
  <0-254>  Maximum number of hops

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by harshitk-arista**
## Change Summary

Support TTL max hops at the peer group and neighbor level

## Related Issue(s)

Fixes #3424

## Component(s) name

`eos_cli_config_gen`

## Proposed changes
CF

## How to test
Molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by philippebureau** [comment]
it works for router_bgp / neighbors 

It is not working for router_bgp / peer_groups

I am getting an error about missing ''ip_address'' attribute
image

i can see the key being added to the structured config properly

image

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 1]
```diff
@@ -171,6 +171,9 @@ router bgp {{ router_bgp.as }}
 {%         if peer_group.ebgp_multihop is arista.avd.defined %}
    neighbor {{ peer_group.name }} ebgp-multihop {{ peer_group.ebgp_multihop }}
 {%         endif %}
+{%         if peer_group.ttl_maximum_hops is arista.avd.defined %}
+   neighbor {{ peer_group.ip_address }} ttl maximum-hops {{ peer_group.ttl_maximum_hops }}
```
Comment (philippebureau) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: neighbor {{ peer_group.name }} ttl maximum-hops {{ peer_group.ttl_maximum_hops }}

**Comment by philippebureau** [comment]
see comment above for proposed fix that works for me

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 (hunk 1 above)
Comment (harshitk-arista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: Yes I pushed out the fix in the last commit

Comment (philippebureau) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: confirmed

**Review by philippebureau — APPROVED** [review]
Tested, LGTM

**Comment by ClausHolbechArista** [comment]
This one needs an update of all molecule scenarios:
```sh
cd ansible_collections/arista/avd/molecule
make refresh-facts
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 2]
```diff
@@ -353,8 +356,12 @@
 {%                     set value_allowas = "Allowed, allowed 3 (default) times" %}
 {%                 endif %}
 {%             endif %}
+{%             set ttl_maximum_hops = "-" %}
+{%             if neighbor.ttl_maximum_hops is arista.avd.defined %}
+{%                 set ttl_maximum_hops = neighbor.ttl_maximum_hops %}
+{%             endif %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]:
```suggestion
{%             set ttl_maximum_hops = neighbor.ttl_maximum_hops | arista.avd.default("-") %}
```

**Review by gmuloc — APPROVED** [review]

**Pull request merged**